app/components/NetworkRender.tsx

```diff
@@ -1,52 +1,31 @@
-import type { ReactElement } from "react";
+import { useNetwork, useSwitchNetwork } from "wagmi";
 import { filteredNetwork } from "~/utils/helpers";
 
-export default function NetworkRender({
-  network,
-  chainName,
-  chainId,
-  switchNetwork,
-  children,
-}: {
-  network: string;
-  chainName?: string;
-  chainId?: number;
-  switchNetwork?: (chainId?: number) => void;
-  children?: ReactElement;
-}) {
-  function getContents() {
-    if (filteredNetwork(network) !== chainName?.toLowerCase()) {
-      if (typeof switchNetwork === "function") {
-        return (
-          <div
-            className={`tw-flex tw-flex-wrap tw-max-w-xs tw-text-sm tw-mx-auto ${children ? "tw-justify-center" : ""}`}
-          >
-            <p className="tw-mb-2"> You are currently connected to {chainName}</p>
-            <button
-              onClick={() => switchNetwork(chainId)}
-              className="tw-bg-[#21C5F2] tw-px-5 tw-py-3 tw-text-sm tw-rounded-lg tw-text-white"
-            >
-              switch network
-            </button>
-          </div>
-        );
-      } else {
-        return (
-          <div
-            className={`tw-flex tw-flex-wrap tw-max-w-xs tw-text-sm tw-mx-auto ${children} ? "tw-justify-center" : ""`}
-          >
-            <p className="tw-mb-2"> You are currently connected to {chainName}</p>
-            <p>Please switch wallet to {filteredNetwork} network</p>
-          </div>
-        );
-      }
-    } else {
-      if (children) {
-        return children;
-      } else {
-        return null;
-      }
-    }
+export default function NetworkRender() {
+  const { chain } = useNetwork();
+  // TODO: switch to other chains?
+  const { switchNetwork } = useSwitchNetwork({ chainId: chain?.id === 1 ? 137 : 1 });
+  const chainName = chain?.name;
+
+  // TODO: Probably don't need this if/else
+  if (typeof switchNetwork === "function") {
+    return (
+      <div className={`tw-flex tw-flex-wrap tw-max-w-xs tw-text-sm tw-mx-auto`}>
+        <p className="tw-mb-2"> You are currently connected to {chainName}</p>
+        <button
+          onClick={() => switchNetwork()}
+          className="tw-bg-[#21C5F2] tw-px-5 tw-py-3 tw-text-sm tw-rounded-lg tw-text-white"
+        >
+          Switch to {chain?.id === 1 ? "Polygon" : "Ethereum Mainnet"}
```

Comment:
I'm sort of confused by this - currently we don't support Ethereum Mainnet so if you're connected to Polygon and that's the chosen network based on the env variable you shouldn't be display a switch button